sonic-net/sonic-buildimage
# [YANG] Update device_metadata to add suppress-fib-pending field

#### Why I did it
 BGP pending FIB suppression is introduced in #12853, hence updating the YANG model.

##### Work item tracking
- Microsoft ADO **(number only)**: 28398135

#### How I did it
Added a new field 'suppress-fib-pending' in the DEVICE_METADATA YANG model table.
Added UT for YANG model changes.

#### How to verify it
Run PR tests.
Apply empty.json file to verify:
```
admin@str2-7050cx3-acs-02:~$ sudo config apply-patch empty.json 
Patch Applier: Patch application starting.
Patch Applier: Patch: []
Patch Applier: Getting current config db.
Patch Applier: Simulating the target full config after applying the patch.
Patch Applier: Validating all JsonPatch operations are permitted on the specified fields
Patch Applier: Validating target config does not have empty tables, since they do not show up in ConfigDb.
Patch Applier: Sorting patch updates.
Patch Applier: The patch was converted into 0 changes.
Patch Applier: Applying 0 changes in order.
Patch Applier: Verifying patch updates are reflected on ConfigDB.
Patch Applier: Patch application completed.
Patch applied successfully.
admin@str2-7050cx3-acs-02:~$ 
```

Signed-off-by: zitingguo-ms <zitingguo@microsoft.com>


#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
